src/EA.Iws.Web/Areas/NotificationMovements/Controllers/ReceiptRecoveryController.cs

```diff
@@ -0,0 +1,366 @@
+﻿namespace EA.Iws.Web.Areas.NotificationMovements.Controllers
+{
+    using Core.Movement;
+    using Core.Shared;
+    using Infrastructure;
+    using Infrastructure.Authorization;
+    using Prsd.Core.Mediator;
+    using Requests.Movement;
+    using Requests.Movement.Complete;
+    using Requests.Movement.Receive;
+    using Requests.Notification;
+    using Requests.NotificationMovements;
+    using Requests.NotificationMovements.Create;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using System.Web;
+    using System.Web.Mvc;
+    using System.Web.Routing;
+    using ViewModels.ReceiptRecovery;
+
+    [AuthorizeActivity(typeof(SaveMovementCompletedReceipt))]
+    public class ReceiptRecoveryController : Controller
+    {
+        private readonly IMediator mediator;
+        private readonly IFileReader fileReader;
+        private const string DateReceivedKey = "DateReceived";
+        private const string DateRecoveredKey = "DateRecovered";
+        private const string UnitKey = "Unit";
+        private const string QuantityKey = "Quantity";
+        private const string ToleranceKey = "Tolerance";
+        private const string CertificateKey = "CertificateType";
+        private const string NotificationTypeKey = "NotificationType";
+        public MovementBasicDetails[] MovementDetails { get; set; }
```

Comment:
I would remove the `MovementDetails` property from the controller and just pass the movement details into the `ValidateShipment` method.